inventory/inventory.go

```diff
@@ -78,34 +77,30 @@ func Get() *InstanceInventory {
 
 	hs := &InstanceInventory{}
 
-	hn, err := os.Hostname()
+	installedPackages, err := packages.GetInstalledPackages()
 	if err != nil {
-		logger.Errorf("os.Hostname() error: %v", err)
+		logger.Errorf("packages.GetInstalledPackages() error: %v", err)
 	}
 
-	hs.Hostname = hn
-
-	di, err := osinfo.GetDistributionInfo()
+	packageUpdates, err := packages.GetPackageUpdates()
 	if err != nil {
-		logger.Errorf("osinfo.GetDistributionInfo() error: %v", err)
+		logger.Errorf("packages.GetPackageUpdates() error: %v", err)
 	}
 
-	hs.LongName = di.LongName
-	hs.ShortName = di.ShortName
-	hs.Version = di.Version
-	hs.KernelVersion = di.Kernel
-	hs.Architecture = di.Architecture
-	hs.OSConfigAgentVersion = config.Version()
-
-	hs.InstalledPackages, err = packages.GetInstalledPackages()
+	oi, err := osinfo.Get()
 	if err != nil {
-		logger.Errorf("packages.GetInstalledPackages() error: %v", err)
+		logger.Errorf("osinfo.Get() error: %v", err)
 	}
 
-	hs.PackageUpdates, err = packages.GetPackageUpdates()
-	if err != nil {
-		logger.Errorf("packages.GetPackageUpdates() error: %v", err)
-	}
+	hs.Hostname = oi.Hostname
```

Comment:
if you include all the members, why not embed osinfo

Reply:
Mainly because of the way the write works. We could change that but that would be a bigger PR